google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/AddFlightsFeed.java

```diff
@@ -0,0 +1,433 @@
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param feedResourceName the resource name of the feed.
+   * @return a Map containing the FlightPlaceholderField and FeedAttribute.
+   */
+  public static Map<FlightPlaceholderField, FeedAttribute> getFeed(
+      GoogleAdsClient googleAdsClient, long customerId, String feedResourceName) {
+    // Constructs the query.
+    String query =
+        "SELECT feed.attributes FROM feed WHERE feed.resource_name = '" + feedResourceName + "'";
+
+    // Constructs the request.
+    SearchGoogleAdsRequest request =
+        SearchGoogleAdsRequest.newBuilder()
+            .setCustomerId(String.valueOf(customerId))
+            .setPageSize(PAGE_SIZE)
+            .setQuery(query)
+            .build();
+
+    // Issues the search request.
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      SearchPagedResponse searchPagedResponse = googleAdsServiceClient.search(request);
+      // Gets the first result because we only need the single feed item we created previously.
+      GoogleAdsRow googleAdsRow = searchPagedResponse.getPage().getResponse().getResults(0);
+      // Gets the attributes list from the feed and creates a map with keys of each attribute and
+      // values of each corresponding ID.
+      List<FeedAttribute> feedAttributeList = googleAdsRow.getFeed().getAttributesList();
+      Map<String, FeedAttribute> feedAttributeMap = new HashMap<>();
+      for (FeedAttribute feedAttribute : feedAttributeList) {
+        feedAttributeMap.put(feedAttribute.getName().getValue(), feedAttribute);
+      }
+
+      // Creates a map to return.
+      Map<FlightPlaceholderField, FeedAttribute> feedAttributes = new HashMap<>();
+
+      // Adds the Flight Description attribute to the map.
+      String flightDescriptionName = "Flight Description";
+      feedAttributes.put(
+          FlightPlaceholderField.FLIGHT_DESCRIPTION, feedAttributeMap.get(flightDescriptionName));
+      // Adds the Destination ID attribute to the map.
+      String destinationIdName = "Destination ID";
+      feedAttributes.put(
+          FlightPlaceholderField.DESTINATION_ID, feedAttributeMap.get(destinationIdName));
+      // Adds the Flight Price attribute to the map.
+      String flightPriceName = "Flight Price";
+      feedAttributes.put(
+          FlightPlaceholderField.FLIGHT_PRICE, feedAttributeMap.get(flightPriceName));
+      // Adds the Flight Sale Price attribute to the map.
+      String flightSalePriceName = "Flight Sale Price";
+      feedAttributes.put(
+          FlightPlaceholderField.FLIGHT_SALE_PRICE, feedAttributeMap.get(flightSalePriceName));
+      // Adds the Final URLs attribute to the map.
+      String finalUrlsName = "Final URLs";
+      feedAttributes.put(FlightPlaceholderField.FINAL_URLS, feedAttributeMap.get(finalUrlsName));
```

Comment:
Add a default case to this, so that if an unexpected feed item is returned it generates an error.